metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+                if remove:
+                    shortestPathEdges.append(tuple((item, item2)))
+        if remove:
+            subGraphSkeleton.remove_edges_from(shortestPathEdges)
+        return length
+
+    def _checkSegmentNotTraced(self, simplePath):
+        """
+        Find if simplePath is not already been visited for sorted segments of a cycle
+
+        Parameters
+        ----------
+        simplePath: list
+           list of nodes in the path
+
+        Returns
+        -------
+        check: Boolean
+            if simplePath is in _sortedSegments check = 0 else check = 1
+
+        Notes
+        -----
+        _sortedSegments: list of lists
+           list of paths already traced
+        """
+        check = 1
+        for path in self._sortedSegments:
+            if len(set(path) & set(simplePath)) > 2:
+                check = 0
+                break
+        return check
+
+    def _singleCycle(self, subGraphSkeleton, cycle):
+        """
+        Find statistics of a single cycle of a disjoint graph
+
+        Parameters
+        ----------
+        subGraphSkeleton: Networkx Graph
+            disjoint subgraph of the skeleton's networkx graph
+
+        cycle: list
+            list of nodes in the cycle
+
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited nodes removed
+
+        cycleInfoDict : Dictionary
+            class instance variable changed in place with key as nth cycle, value as list with
+            [number of branch points on cycle, length of the cycle]
+
+        countDict: Dictionary
+            class instance variable changed in place with key as source node,
+            value is the number of segments attached to the source node
+
+        lengthDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the length of the segment
+
+        tortuosityDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the tortuosity of the segment (Nan but saved as zero)
+
+        contractionDictdict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the contraction of the segment (0) since it is a cycle
+        """
+        sourceOnCycle = cycle[0]
+        if sourceOnCycle not in self._visitedSources:
+            self.countDict[sourceOnCycle] = 1
+            self._visitedSources.append(sourceOnCycle)
+        else:
+            self.countDict[sourceOnCycle] += 1
+        curveLength = self._getLengthAndRemoveTracedPath(subGraphSkeleton, cycle, 1)
+        self.cycleInfoDict[self.cycles] = [0, curveLength]
```

Comment:
So, does this dictionary entry get overwritten each time the function runs, or does self.cycles get incremented somewhere else?